packages/devtools_app/lib/src/inspector/flutter/inspector_tree_flutter.dart

```diff
@@ -83,18 +84,21 @@ class _InspectorTreeRowState extends State<_InspectorTreeRowWidget>
 
 class InspectorTreeControllerFlutter extends Object
     with InspectorTreeController, InspectorTreeFixedRowHeightController {
+  // Controller for controlling custom animation in the Tree
+  AnimationController animationController;
+
   /// Client the controller notifies to trigger changes to the UI.
-  InspectorControllerClient get client => _client;
-  InspectorControllerClient _client;
-  set client(InspectorControllerClient value) {
-    if (_client == value) return;
-    // Do not set a new client if there is still an old client.
-    assert(value == null || _client == null);
-    _client = value;
-
-    if (config.onClientActiveChange != null) {
-      config.onClientActiveChange(value != null);
-    }
+  List<InspectorControllerClient> _clients = [];
+
+  void addClient(InspectorControllerClient client) {
```

Comment:
this change is required since I need LayoutDetailsTab to listen on the Summary tree controller now.
I tried listening to the InspectorService directly but I couldn't figure out stuffs

Reply:
you should use inspector_controller rather than inspector_service.
The inspector_controller will give you a natural place to listen for changes to the inspector without getting too coupled to controlling a specific inspector tree.